src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -222,6 +232,39 @@ private static GitPullRequest CreatePullRequest(string sourceBranch, string targ
             throw new InvalidOperationException("Could not find drop path");
         }
 
+        private static async Task<string> DownloadArtifactsAsync(BuildHttpClient buildClient, Build build, BuildArtifact artifact, CancellationToken cancellationToken)
+        {
+            var tempDirectory = Path.Combine(Path.GetTempPath(), string.Concat(Options.InsertionName, Options.BranchName).Replace(" ", "_").Replace("/","_"));
+            if (Directory.Exists(tempDirectory))
+            {
+                // Be judicious and clean up old artifacts so we do not eat up memory on the scheduler machine.
+                // Sometime FileSystem APIs do not create a directory immediately after a deletion - Looks like there is a race.
+                // So I'm just cleaning up the old directories contents in this case.
+                foreach (var dir in Directory.GetDirectories(tempDirectory)) Directory.Delete(dir, recursive: true);
```

Comment:
Why not just delete the entire tempDirectory at once?

Reply:
Added a comment in code on why I had to do what I did.

Reply:
I couldn't find any documentation for why this happens though, just observations from [here](https://stackoverflow.com/questions/32593552/using-directory-delete-and-directory-createdirectory-to-overwrite-a-folder) which I ran into during testing. Would be happy to try other suggestions out though.